### PR TITLE
Add support for default merge commit params

### PR DESCRIPTION
**Pull request by garrettheel**
Alternative to https://github.com/integrations/terraform-provider-github/pull/1201, adds support for:

 * `squash_merge_commit_title`
 * `squash_merge_commit_message`
 * `merge_commit_title`
 * `merge_commit_message`

~Note: there are currently some build errors with `github.DependabotEncryptedSecret` that depend on https://github.com/integrations/terraform-provider-github/pull/1258. Will rebase after that is merged.~

### CODE REVIEW

File: github/data_source_github_repository.go [hunk 1]
```diff
@@ -234,6 +250,11 @@ func dataSourceGithubRepositoryRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("allow_squash_merge", repo.GetAllowSquashMerge())
 	d.Set("allow_rebase_merge", repo.GetAllowRebaseMerge())
 	d.Set("allow_auto_merge", repo.GetAllowAutoMerge())
+	d.Set("allow_auto_merge", repo.GetAllowAutoMerge())
```
Comment (JTarasovic) [inline github/data_source_github_repository.go:253]: This seems to be duplicated.

```suggestion
```

Comment (garrettheel) [reply github/data_source_github_repository.go]: thanks for catching - fixed!

**Comment by gesellix** [comment]
@garrettheel #1258 has been merged.

**Comment by kfcampbell** [comment]
Once this PR is rebased, I can merge/release!

**Comment by garrettheel** [comment]
@kfcampbell I just rebased and bumped go-github up to v47

**Review by kfcampbell — APPROVED** [review]
Integration tests for these items are failing on main and failing here in the exact same way, so it's not a regression! 

Thanks for doing this.

**Comment by chaosaffe** [comment]
It looks like this change may not have considered repos that have `merge` and `squash` disabled (i.e. `rebase` only). It appears there is no way to _not_ set these values, as `""` is not a valid value, and nulling them applies the defaults. 

I'm happy to throw up a PR, but first wanted to ask, is there a preferred way to patch this?

**Comment by kfcampbell** [comment]
@chaosaffe Hmm...I'm not sure! I don't know off the top of my head, sorry.

**Comment by garrettheel** [comment]
@chaosaffe would you mind creating a new issue to discuss? I'm curious what the negative impact to you in setting those defaults is. I also don't know the answer off the top of my head, but happy to look into it with you

**Comment by danielpops** [comment]
Perhaps an answer to the above question(s), with the shipping of this PR, my terraform plans suddenly show this diff for every repo in my organization:
```
  # github_repository.repo["the_repo"] will be updated in-place
  ~ resource "github_repository" "repo" {
        id                          = "the_repo"
      + merge_commit_message        = "PR_TITLE"
      + merge_commit_title          = "MERGE_MESSAGE"
        name                        = "the_repo"
      + squash_merge_commit_message = "COMMIT_MESSAGES"
      + squash_merge_commit_title   = "COMMIT_OR_PR_TITLE"
        # (28 unchanged attributes hidden)
}
```

This does not seem useful or expected for me, and I imagine it may confuse others.

**Comment by garrettheel** [comment]
@danielpops @chaosaffe I created https://github.com/integrations/terraform-provider-github/issues/1295 to track your concerns. Please use that to discuss.

**Pull request merged**